zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientRequestReceiver.java

```diff
@@ -269,12 +267,13 @@ private HttpRequestMessage buildZuulHttpRequest(
 
         // This is the only way I found to get the port of the request with netty...
         final int port = channel.attr(SourceAddressChannelHandler.ATTR_SERVER_LOCAL_PORT).get();
-        final HAProxyMessage haProxyMessage = channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_MESSAGE).get();
-        if (haProxyMessage != null) {
-            context.set(CommonContextKeys.PROXY_PROTOCOL_PORT, haProxyMessage.destinationPort());
-        }
         final String serverName = channel.attr(SourceAddressChannelHandler.ATTR_SERVER_LOCAL_ADDRESS).get();
         final SocketAddress clientDestinationAddress = channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDR).get();
```

Comment:
@argha-c confusing as the name is, in this case `ATTR_LOCAL_ADDR` is in fact the client destination address. See the test line I added to `ClientReceiverRequestTest`:

`assertEquals((int) result.getClientDestinationPort().get(), hapmDestinationAddress.getPort());`

Reply:
Ack